src/main/kotlin/uk/gov/justice/digital/hmpps/incentivesapi/jpa/repository/PrisonIncentiveLevelRepository.kt

```diff
@@ -97,4 +97,17 @@ interface PrisonIncentiveLevelRepository : CoroutineCrudRepository<PrisonIncenti
     """,
   )
   fun findPrisonIdsWithActiveLevels(): Flow<String>
+
+  /**
+   * All prisons that have this level active
+   */
+  @Query(
+    // language=postgresql
+    """
+    SELECT prison_id
+    FROM prison_incentive_level
+    WHERE active IS TRUE AND level_code = :levelCode
+    """,
+  )
+  fun findPrisonIdsWithActiveLevel(levelCode: String): Flow<String>
```

Comment:
Could this just count the rows, we don't need the IDs don't we?

Reply:
right now, you're right but i'm anticipating that i'll need the ids for future error message shown to users

Reply:
Don't mind too much in this case, I don't think it would be a big list.